addons/l10n_sa_edi/models/account_move.py

```diff
@@ -210,6 +210,15 @@ def _l10n_sa_is_in_chain(self):
         zatca_doc_ids = self.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'sa_zatca')
         return len(zatca_doc_ids) > 0 and not any(zatca_doc_ids.filtered(lambda d: d.state == 'to_send'))
 
+    def _get_tax_lines_to_aggregate(self):
+        """
+        If the final invoice has downpayment lines, we skip the tax correction, as we need to recalculate tax amounts
+        without taking into account those lines
+        """
+        if self.country_code == 'SA' and not self._is_downpayment() and self.line_ids._get_downpayment_lines():
+            return self.env['account.move.line']
```

Comment:
This seems brutal.  So, skip the rounding in the entire invoice if any downpayment line?

Reply:
Yes, the current solution is to skip, in this specific case, the ["recently" introduced](https://github.com/odoo/odoo/commit/b5ea436240604061276df75d921f84077b11129b) step to address quick-encoding or manual modification of the line. 
For a more fine grained approach we could:
- keep all the base lines in `to_process` (postpone `filter_tax_values_to_apply`)
- make the tax adjustment with the full unfiltered amounts
- apply the invoice line filter in `to_process` just before calling `_aggregate_taxes`

However, it feels we are adapting a generic method to this specific case, we don't know in advance what `filter_tax_values_to_apply` does and why it is needed.

Reply:
At least the change is limited to the customer invoice cases with downpayment lines.